packages/federation/src/managed-federation.ts

```diff
@@ -0,0 +1,193 @@
+import { GraphQLSchema } from 'graphql';
+import type { FetchFn } from '@graphql-tools/executor-http';
+import { ExecutionResult } from '@graphql-tools/utils';
+import { fetch as defaultFetch } from '@whatwg-node/fetch';
+import {
+  getStitchedSchemaFromSupergraphSdl,
+  GetSubschemasFromSupergraphSdlOpts,
+} from './supergraph.js';
+
+export type FetchSupergraphSdlFromManagedFederationOpts = {
+  /**
+   * The graph ID of the managed federation graph.
+   */
+  graphId: string;
+  /**
+   * The API key to use to authenticate with the managed federation up link.
+   * It needs at least the `service:read` permission.
+   */
+  apiKey: string;
+  /**
+   * The URL of the managed federation up link.
+   * Default: 'https://uplink.api.apollographql.com/' (Apollo's managed federation up link on GCP)
+   * Alternative: 'https://aws.uplink.api.apollographql.com/' (Apollo's managed federation up link on AWS)
+   */
+  upLink?: string;
+  /**
+   * The ID of the last fetched supergraph.
+   * If provided, a supergraph is returned only if the managed supergraph have changed.
+   */
+  lastSeenId?: string;
+  /**
+   * The fetch implementation to use.
+   * Default: global.fetch
+   */
+  fetch?: FetchFn;
+};
+
+export type RouterConfig = {
+  supergraphSdl: string;
+  minDelaySeconds: number;
+  id: string;
+};
+
+export type Unchanged = {
+  minDelaySeconds?: never;
+  id: string;
+};
+
+export type FetchError = {
+  error: { code: string; message: string };
+  minDelaySeconds: never;
+};
+
+type RouterConfigResult = {
+  routerConfig:
+    | ({ __typename: 'RouterConfigResult' } & RouterConfig)
+    | ({ __typename: 'Unchanged' } & Unchanged)
+    | { __typename: 'FetchError'; code: string; message: string; minDelaySeconds: never };
+};
+
+export const DEFAULT_UPLINK_ENDPOINTS = [
+  'https://uplink.api.apollographql.com/',
+  'https://aws.uplink.api.apollographql.com/',
```

Comment:
@ardatan Why do we want an array of default uplinks if we actually use only one ?